Jenkinsfile

```diff
@@ -84,11 +88,39 @@ pipeline {
                                 }
                                 build.untilEach {
                                     // Wait until a build object is complete
-                                    return it.object().status.phase == "Complete"
+                                    def phase = it.object().status.phase
+
+                                    if(phase == "Complete") {
+                                        return true
+                                    } else if( phase == "Failed" ) {
+                                        currentBuild.result = 'FAILURE'
+                                        return true
+                                    }
+                                    else {
+                                        return false
+                                    }
+                                }
+                            }
+                            if(currentBuild.result == 'FAILURE') {
+                                error("some failure")
```

Comment:
We probably can't point to the exact cause of the failure, but maybe the error message can be a bit more informative? like point to the build logs...

Reply:
@codificat yeah I can print the logs.  Will add that.